src/Zicht/Bundle/UrlBundle/Aliasing/Listener.php

```diff
@@ -147,76 +168,102 @@ protected function isExcluded($url)
      */
     public function onKernelRequest(Event\GetResponseEvent $event)
     {
-        if ($event->getRequestType() === HttpKernelInterface::MASTER_REQUEST) {
-            $request = $event->getRequest();
-            $publicUrl = rawurldecode($request->getRequestUri());
+        if ($event->getRequestType() !== HttpKernelInterface::MASTER_REQUEST) {
+            return;
+        }
 
-            if ($this->isExcluded($publicUrl)) {
-                // don't process urls which are marked as excluded.
-                return;
-            }
+        $request = $event->getRequest();
+        $publicUrl = rawurldecode($request->getRequestUri());
 
-            if ($this->isParamsEnabled) {
-                if (false !== ($queryMark = strpos($publicUrl, '?'))) {
-                    $originalUrl = $publicUrl;
-                    $publicUrl = substr($originalUrl, 0, $queryMark);
-                    $queryString = substr($originalUrl, $queryMark);
-                } else {
-                    $queryString = null;
-                }
+        if ($this->isExcluded($publicUrl)) {
+            // don't process urls which are marked as excluded.
+            return;
+        }
 
-                $parts = explode('/', $publicUrl);
-                $params = array();
-                while (false !== strpos(end($parts), '=')) {
-                    array_push($params, array_pop($parts));
-                }
-                if ($params) {
-                    $publicUrl = join('/', $parts);
+        $queryString = '';
+        if (false !== ($queryMark = strpos($publicUrl, '?'))) {
+            $queryString = substr($publicUrl, $queryMark);
+            $publicUrl = substr($publicUrl, 0, $queryMark);
```

Comment:
(flavor) zou regel 185 en 186 omdraaien.  Logischer om eerst het begin en daarna het einde van de string te doen

Reply:
Klopt, alleen dan kun je `$publicUrl` URL niet meer gebruiken, want die is al gestript :wink: